presto-common/src/main/java/com/facebook/presto/common/block/EncoderUtil.java

```diff
@@ -97,4 +100,41 @@ public static void encodeNullsAsBits(SliceOutput sliceOutput, Block block)
 
         return Optional.of(valueIsNull);
     }
+
+    /**
+     * Decode the bit stream created by encodeNullsAsBits.  The valueIsNull buffer must be exactly sized to the position
+     * count.  Returns the number of nulls seen during deserialization.
+     */
+    public static int decodeNullBits(SliceInput sliceInput, boolean[] valueIsNull)
```

Comment:
Consider refactoring to avoid so much copy-paste between this new version and existing one.